x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -153,7 +153,7 @@ private Connector(
         this.error = error;
         this.features = features;
         this.filtering = Objects.requireNonNull(filtering, "[filtering] cannot be null");
-        this.indexName = Objects.requireNonNull(indexName, "[index_name] cannot be null");
```

Comment:
As `indexName` is annotated with `@Nullable` I think we should not have this check. (led to a failure in one of the new test cases, so I simply fixed it directly with this PR).